src/search/operator_counting/delete_relaxation_constraints.cc

```diff
@@ -0,0 +1,241 @@
+}
+
+int DeleteRelaxationConstraints::get_constraint_id(FactProxy f) {
+    return constraint_ids[f.get_variable().get_id()][f.get_value()];
+}
+
+void DeleteRelaxationConstraints::create_auxiliary_variables(
+    TaskProxy task_proxy, named_vector::NamedVector<lp::LPVariable> &variables) {
+    OperatorsProxy ops = task_proxy.get_operators();
+    int num_ops = ops.size();
+    VariablesProxy vars = task_proxy.get_variables();
+    int num_vars = vars.size();
+
+    // op_used
+    add_lp_variables(num_ops, variables, lp_var_id_op_used, 0, 1, 0, use_integer_vars);
+
+    // fact_reached
+    lp_var_id_fact_reached.resize(num_vars);
+    for (VariableProxy var : vars) {
+        add_lp_variables(var.get_domain_size(), variables,
+                         lp_var_id_fact_reached[var.get_id()],
+                         0, 1, 0, use_integer_vars);
+    }
+
+    // first_achiever
+    lp_var_id_first_achiever.resize(num_ops);
+    for (OperatorProxy op : ops) {
+        lp_var_id_first_achiever[op.get_id()].resize(num_vars);
+        for (VariableProxy var : vars) {
+            add_lp_variables(var.get_domain_size(), variables,
+                             lp_var_id_first_achiever[op.get_id()][var.get_id()],
+                             0, 1, 0, use_integer_vars);
+        }
+    }
+
+    if (use_time_vars) {
+        // op_time
+        add_lp_variables(num_ops, variables, lp_var_id_op_time, 0, num_ops, 0, use_integer_vars);
+
+        // fact_time
+        lp_var_id_fact_time.resize(num_vars);
+        for (VariableProxy var : vars) {
+            add_lp_variables(var.get_domain_size(), variables,
+                             lp_var_id_fact_time[var.get_id()],
+                             0, num_ops, 0, use_integer_vars);
+        }
+    }
+}
+
+void DeleteRelaxationConstraints::create_constraints(TaskProxy task_proxy,
+    lp::LinearProgram &lp) {
+    named_vector::NamedVector<lp::LPVariable> &variables = lp.get_variables();
+    named_vector::NamedVector<lp::LPConstraint> &constraints = lp.get_constraints();
+    double infinity = lp.get_infinity();
+    OperatorsProxy ops = task_proxy.get_operators();
+    VariablesProxy vars = task_proxy.get_variables();
+
+    // All goal facts must be reached (handled in variable bound instead of constraint).
+    // R_f = 1 for all goal facts f.
+    for (FactProxy goal : task_proxy.get_goals()) {
+        variables[get_var_fact_reached(goal)].lower_bound = 1;
+    }
+
+    // A fact is reached if it has a first achiever or is true in the current state.
+    // sum_{o \in achievers(f)} F_{o,f} - R_f >= [s |= f] for each fact f.
+    constraint_ids.resize(vars.size());
+    for (VariableProxy var : vars) {
+        constraint_ids[var.get_id()].resize(var.get_domain_size());
+        for (int value = 0; value < var.get_domain_size(); ++value) {
+            constraint_ids[var.get_id()][value] = constraints.size();
+            constraints.emplace_back(0, infinity);
+            /* The constraint is:
```

Comment:
Did you mean to add something here?

Reply:
This is old code, I don't remember. I removed this line and only left the comment below. The comment a few lines up explains the constraint, so I think this is fine.